src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -642,5 +674,84 @@ public static ClusterState readFrom(StreamInput in, @Nullable DiscoveryNode loca
             }
             return builder.build();
         }
+
+    }
+
+    public Diff diffs(ClusterState previousState) {
+        return new ClusterStateDiff(previousState, this);
+    }
+
+
+    private class ClusterStateDiff implements Diff {
+
+        private final long version;
```

Comment:
Indeed redundant but clear (I had to double check because I wasn't aware of the convention, as most people reading this will be). I suggested the `to` & `from` to make it shorter so less of an overheard. as said, just a suggestion.